cli/command/environment.go

```diff
@@ -217,42 +231,47 @@ func (e *EnvironmentCommand) update(c *cli.Context) error {
 }
 
 func (e *EnvironmentCommand) link(c *cli.Context) error {
-	args, err := extractArgs(c.Args(), "SOURCE_ENVIRONMENT_NAME", "DESTINATION_ENVIRONMENT_NAME")
-	if err != nil {
-		return err
-	}
+	generateAddLinkRequest := func(sourceEnvironmentID, destEnvironmentID string) (models.UpdateEnvironmentRequest, error) {
+		env, err := e.client.ReadEnvironment(sourceEnvironmentID)
+		if err != nil {
+			return models.UpdateEnvironmentRequest{}, err
+		}
 
-	sourceEnvironmentID, err := e.resolveSingleEntityIDHelper("environment", args["SOURCE_ENVIRONMENT_NAME"])
-	if err != nil {
-		return err
-	}
+		if !client.Contains(destEnvironmentID, env.Links) {
+			appendedLinks := append(env.Links, destEnvironmentID)
```

Comment:
Wait, I'm a little confused here. Does this build?

```
if !client.Contains(destEnvironmentID, env.Links) {
    appendedLinks := append(env.Links, destEnvironmentID)
}

req := models.UpdateEnvironmentRequest{Links: &appendedLinks}
```

since `appendedLinks` is declared within the `if` statement, isn't is out of scope when you try to reference it in `models.UpdateEnvironmentRequest`?

Reply:
woah - this is really bad. I thought I was making a minor change (apparently not). Fixed! ✅